ompi/mca/part/base/part_base_frame.c

```diff
@@ -0,0 +1,153 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2004-2010 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2007 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2007 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2009      Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2013-2015 Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2015      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * Copyright (c) 2018      IBM Corporation. All rights reserved.
+ * Copyright (c) 2020      Intel, Inc.  All rights reserved.
+ * Copyright (c) 2020      Sandia National Laboratories. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+
+#include "ompi_config.h"
+#include <stdio.h>
+
+#include <string.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif  /* HAVE_UNIST_H */
+#include "ompi/mca/mca.h"
+#include "opal/util/output.h"
+#include "opal/mca/base/base.h"
+
+
+#include "ompi/constants.h"
+#include "ompi/mca/part/part.h"
+#include "ompi/mca/part/base/base.h"
+#include "ompi/mca/part/base/part_base_prequest.h"
+
+/*
+ * The following file was created by configure.  It contains extern
+ * statements and the definition of an array of pointers to each
+ * component's public mca_base_component_t struct.
+ */
+
+#include "ompi/mca/part/base/static-components.h"
+
+/*
+ * This function allows modules to not declare a progress function,
+ * by defaulting to doing nothing.
+ */
+int mca_part_base_progress(void)
+{
+    return OMPI_SUCCESS;
+}
+
+#define xstringify(part) #part
+#define stringify(part) xstringify(part)
+
+/*
+ * Global variables
+ */
+mca_part_base_module_t mca_part = {
+    mca_part_base_progress,   /* part_progress */
```

Comment:
Designated initializers are strongly recommended here. You can drop all the NULLs as the language guarantees uninitialized members are 0.